### PR TITLE
Diagonal move refactor electric boogaloo

### PR DESCRIPTION
**Pull request by BraveMole**
## About The Pull Request

Mostly code improvement, as it's clearer to read and the whole move proc is probably more performant

Tested so far:
 - You cannot phase between two walls while moving diagonally
 - You cannot phase between two mobs if you weren't able to bump/shuffle either
 - You cannot phase throught baricades or windows
 - Pulling seems to have normal behaviour
 - Push seems to have normal behaviour (you can now push diagonally as a bonus)
 - Shuffling seems to have normal behaviour (You can now shuffle diagonally as a bonus)

https://github.com/tgstation/tgstation/pull/62605

## Why It's Good For The Game

Better perf, move is the most expensive proc by far

## Changelog
:cl:
balance: You can push/shuffle diagonally
/:cl:

### CODE REVIEW

**Comment by BraveMole** [comment]
DNTM pending tg review

**Comment by BraveMole** [comment]
Kyler said cool

File: code/__DEFINES/dcs/signals.dm [hunk 1]
```diff
@@ -177,7 +177,6 @@
 
 
 // /atom/movable signals
-#define COMSIG_MOVABLE_PRE_MOVE "movable_pre_move"				//from base of atom/movable/Moved(): (/atom)
```
Comment (BraveMole) [inline code/__DEFINES/dcs/signals.dm:180]: not used anywhere

File: code/modules/admin/smites/puzzle.dm [hunk 2]
```diff
@@ -216,17 +216,17 @@
 	var/icon/puzzle_icon
 
 /obj/structure/puzzle_element/Move(nloc, dir)
-	if(!isturf(nloc) ||  moving_diagonally || get_dist(get_step(src,dir),get_turf(source)) > 1)
+	if(!isturf(nloc) || (dir - 1) & dir|| get_dist(get_step(src,dir),get_turf(source)) > 1)
```
Comment (BraveMole) [inline code/modules/admin/smites/puzzle.dm:219]:
```suggestion
	if(!isturf(nloc) || (dir - 1) & dir || get_dist(get_step(src,dir),get_turf(source)) > 1)
```

**Comment by TiviPlus** [comment]
can you link the tg pr btw

**Comment by Nannaquin** [comment]
Bore witness to possibly being diag tossed through a cade by a warrior. Unsure if this is normal or a bug.

**Comment by BraveMole** [comment]
> Bore witness to possibly being diag tossed through a cade by a warrior. Unsure if this is normal or a bug.

That's normal if the cade was not wired

**Comment by Nannaquin** [comment]
> 
> 
> > Bore witness to possibly being diag tossed through a cade by a warrior. Unsure if this is normal or a bug.
> 
> That's normal if the cade was not wired

shoulda mentioned it was wired.

**Comment by novaepee** [comment]
holy smokes, this got merged

**Pull request merged**